addons/account/models/account_move.py

```diff
@@ -3498,11 +3498,22 @@ def _post(self, soft=True):
             if wrong_lines:
                 wrong_lines.write({'partner_id': invoice.commercial_partner_id.id})
 
+        # reconcile if state is in draft and move has reversal_entry_id set
+        draft_refunds = to_post.filtered(lambda refund: refund.reversed_entry_id and refund.state == 'draft')
```

Comment:
```suggestion
        draft_reverse_moves = to_post.filtered(lambda move: move.reversed_entry_id)
```
It could also be journal entries (so not a refund).
And you don't need to check the state, it cannot be posted.
Though it could technically be cancel, we should also change the check done above (line 3459) to include cancelled moves.